### PR TITLE
completed hometask1 nikolayenko

### PR DESCRIPTION
**Pull request by mennekis**
completed hometask1 Nikolayenko

### CODE REVIEW

**Comment by mennekis** [comment]
Hi there) here is the Hometask_2 from the Oleksiy  Nikolayenko

File: Hometask_2/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    price: 900,
+    condition: "new",
+  };
+
+  function showProduct(product: Tproduct) {
+    console.log(product);
+  }
+
+  showProduct(product);
+  showProduct(product2 as Tproduct);
```
Comment (lyubomyr-rudko) [inline Hometask_2/src/index.ts:139]: no need for explicit casting here

File: Hometask_2/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  function incrementsAge(person: TPerson): void {
+    person.age = person.age + 1;
+  }
+  incrementsAge(person);
+  console.log(`Name: ${person.name}, Age: ${person.age}`);
+}
+excercise6();
+
+function excercise7() {
+  function map(arr: number[], fn: (num: number) => number): number[] {
```
Comment (lyubomyr-rudko) [inline Hometask_2/src/index.ts:83]: Some other way to implement this is to create type annotation for this function and use it here and in line 94

File: Hometask_2/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    y: number;
+  };
+
+  const p1: TPoint = { x: 1, y: 1 };
+  const p2: TPoint = { x: 4, y: 5 };
+
+  function distance(p1: TPoint, p2: TPoint): number {
+    const { x: x1, y: y1 } = p1;
+    const { x: x2, y: y2 } = p2;
+    const deltaX = x2 - x1;
```
Comment (lyubomyr-rudko) [inline Hometask_2/src/index.ts:40]: dx an dy names actually makes more sense here than verbose `deltaX`

**Pull request closed without merging**